rust/lance/src/encodings/plain.rs

```diff
@@ -209,6 +209,7 @@ impl<'a> PlainDecoder<'a> {
                     "PlainDecoder: request([{}..{}]) out of range: [0..{}]",
                     start, end, self.length
                 ),
+                location: location!(),
```

Comment:
Can `location!()` to be in a helper function / macro?  Calling it everywhere is a bit tedious.